src/nvim/ugrid.c

```diff
@@ -0,0 +1,137 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <limits.h>
+
+#include "nvim/vim.h"
+#include "nvim/ui.h"
+#include "nvim/ugrid.h"
+
+#ifdef INCLUDE_GENERATED_DECLARATIONS
+# include "ugrid.c.generated.h"
+#endif
+
+void ugrid_init(UGrid *grid)
+{
+  grid->attrs = EMPTY_ATTRS;
+  grid->fg = grid->bg = -1;
+  grid->cells = NULL;
+}
+
+void ugrid_free(UGrid *grid)
+{
+  xfree(grid->cells);
```

Comment:
What about the different grid rows? Don't we need to call `destroy_cells(grid);` here?